pytorch_to_returnn/torch/nn/utils/rnn.py

```diff
@@ -1,15 +1,31 @@
 import numpy
+from ..functional import full
 from ..modules.rnn import PackedSequence
 from ..modules.shape import FlattenBatch
 from ...tensor import Tensor
 from ..._C import from_numpy
+from .... import torch
 from ....naming import Naming
 
 
+def _batch_sizes_from_lengths(lengths: Tensor) -> Tensor:
+  indices = torch.arange(torch.max(lengths))  # [T]
+  indices_bc = indices.unsqueeze(0)  # [1,T]
+  l_bc = lengths.unsqueeze(1)  # [B,1]
+  mask = indices_bc < l_bc  # [B,T]
```

Comment:
The broadcasting here seems not to work yet.

Reply:
If there is some issue specifically about broadcasting (not related to previous other problems), then you should make a separate test case only for the broadcasting, and also fix that.

Reply:
Maybe you mean this from the failing test case:
```
  File "/home/runner/work/pytorch-to-returnn/pytorch-to-returnn/pytorch_to_returnn/torch/nn/utils/rnn.py", line 15, in _batch_sizes_from_lengths
    line: mask = indices_bc < l_bc  # [B,T]
    locals:
      mask = <not found>
      indices_bc = <local> <Tensor name:? tensor:(1,5) returnn_data:'Unflatten_output' [F|'Unflatten_split_dims0'(1),T|'Range_input_len'[]] axes id>
      l_bc = <local> <Tensor name:? tensor:(Batch(3),1) returnn_data:'Unflatten_1_output' [B,F|'Unflatten_1_split_dims1'(1)] axes id>
```
In the following back trace, there is some problem in `_unify_tensor_axes_returnn_meta`. It creates a `ReinterpretDataLayer` with `'set_axes': {'F': 'B'}`, which does not make sense (and that is also the error then).
We should make a separate test case just reproducing the problem here.

Reply:
In general, `_unify_tensor_axes_returnn_meta` should not reassign any axis ("F" here) to the batch axis, or neither the other way around (reassign the batch axis ("B") to some other axis). Maybe if you just exclude this (reassignments on/to the batch axis), it's already fine.

Reply:
I added a test case `test_broadcasting_with_lengths`. However, even if I avoid reassigning the batch axis, the problem persists. Should I anyway push a commit to avoid reassigning the batch axis?

Reply:
This is fixed now.